cdc/sorter/leveldb/system/system.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package system
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"hash/fnv"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/sorter"
+	lsorter "github.com/pingcap/ticdc/cdc/sorter/leveldb"
+	"github.com/pingcap/ticdc/pkg/actor"
+	"github.com/pingcap/ticdc/pkg/actor/message"
+	"github.com/pingcap/ticdc/pkg/config"
+	cerrors "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/syndtr/goleveldb/leveldb"
+	"go.uber.org/zap"
+)
+
+// The interval of collecting leveldb metrics.
+const defaultMetricInterval = 15 * time.Second
+
+// State of a system.
+type sysState int
+
+const (
+	sysStateInit    sysState = 0
+	sysStateStarted sysState = 1
+	sysStateStopped sysState = 2
+)
+
+// System manages leveldb sorter resource.
+type System struct {
+	dbs         []*leveldb.DB
+	dbSystem    *actor.System
+	dbRouter    *actor.Router
+	cleanSystem *actor.System
+	cleanRouter *actor.Router
+	cfg         *config.SorterConfig
+	closedCh    chan struct{}
+	closedWg    *sync.WaitGroup
+
+	state   sysState
+	stateMu *sync.Mutex
+}
+
+// NewSystem returns a system.
+func NewSystem(cfg *config.SorterConfig) *System {
+	dbSystem, dbRouter := actor.NewSystemBuilder("sorter").
+		WorkerNumber(cfg.LevelDB.Count).Build()
+	cleanSystem, cleanRouter := actor.NewSystemBuilder("cleaner").
+		WorkerNumber(cfg.LevelDB.Count).Build()
+	return &System{
+		dbSystem:    dbSystem,
+		dbRouter:    dbRouter,
+		cleanSystem: cleanSystem,
+		cleanRouter: cleanRouter,
+		cfg:         cfg,
+		closedCh:    make(chan struct{}),
+		closedWg:    new(sync.WaitGroup),
+		state:       sysStateInit,
+		stateMu:     new(sync.Mutex),
+	}
+}
+
+// GetActorID returns an ActorID correspond with tableID.
+func (s *System) GetActorID(tableID uint64) actor.ID {
```

Comment:
Prefer to use the method name `ActorID` because it's more idiomatically in Golang. Ref: https://golang.org/doc/effective_go#Getters